module/zfs/arc.c

```diff
@@ -881,6 +881,14 @@ static inline void arc_hdr_clear_flags(arc_buf_hdr_t *hdr, arc_flags_t flags);
 static boolean_t l2arc_write_eligible(uint64_t, arc_buf_hdr_t *);
 static void l2arc_read_done(zio_t *);
 
+/*
+ * l2arc_mfuonly : A ZFS module parameter that controls whether only MFU
+ * 		metadata and data are cached from ARC into L2ARC. This reduces
+ * 		the wear of L2ARC devices and may be beneficial in certain
+ * 		workloads.
```

Comment:
Modern SSDs won't have an endurance issue as cache devices. You need a better reason here.
Also, vaguely referring to "certain workloads" is not helpful for anyone deciding whether or not to enable the feature.

How about adding a method in the man page (missing from this PR!) to describe how to observe
your current workload's MFU size, l2arc feed rate, and eviction rate for eligible blocks as a method
for inferring if MFU-only makes sense for this workload. In other words, if the cache is 2TiB and your MFU is 100MiB, then maybe you don't want to set this parameter.

Reply:
This is a great suggestion. The more obvious use case would be in case of a `zfs send`, so that L2ARC space is not wasted by filling it with MRU data/metadata.

Reply:
@richardelling a very simple workload which is impacted by current L2ARC behavior is backup (ie: sequential read) of big files (think about vm disk image): when a L2ARC device is used, the high "churn rate" of ARC MFU means that reading will cause many L2ARC writes, wearing off the device. While it is true that mixed-use enterprise disks have significant endurance rating, one should not be limited to (relatively) expensive disk for an L2ARC device. For a real-world example, I have a server with 2x striped Intel S4601 480 GB as L2ARC and, when copying virtual machine files, I often literally "burn" one entire SSD daily write. While the L2ARC parameters where set to more aggressive level (ie: increased `l2arc_headroom` and `l2arc_write_max`), it seems very wrong to wear the SSDs when *reading* a big file.

Reply:
Document your use case and put some numbers on it. For example, I can't easily google the specs on that drive, but an Intel DC S3520 480 GB SSD has an endurance rating of 945TBW. This equates to a full drive write every day for 5.4 years. NB, since we're measuring actual drive writes, you can just use the measured amount of writes. Also note, the warranty is 5 years, so expect to replace in that timeframe. So now you have your use case workload defined. But that workload may or may not be cause for concern. For your described workload with the above specs, the change is not justified. But for other workloads and specs, it might be justified... help out the user here.

Reply:
@richardelling well, if the change only adds a potentially useful option *and* it is not invasive/difficult to maintain, I think it would be useful. As said above, enterprise SSD are quite durable. However, for L2ARC it shoult be reasonable (for some cases at least) to use cheaper, even consumer grade SSD which, while having lower write performance/endurance, often are quite fast for reads. Finally, consider than having an L2ARC with potentially very useful psedo-MFU data being trashed by a sequential copy with MFU data is going to lower system performance. Giving the user the possibility of doing an informed choice does not seems bad to me, unless it is a maintenance burden.